packages/block-library/src/navigation/edit/index.js

```diff
@@ -490,11 +549,18 @@ function Navigation( {
 					isResolvingCanUserCreateNavigationMenu={
 						isResolvingCanUserCreateNavigationMenu
 					}
-					onFinish={ ( post ) => {
-						if ( post ) {
-							setRef( post.id );
+					onFinish={ ( navPostOrClassicMenu ) => {
+						const isClassicMenu = navPostOrClassicMenu.hasOwnProperty(
+							'auto_add'
+						);
+						if ( isClassicMenu ) {
+							convert(
+								navPostOrClassicMenu.id,
+								navPostOrClassicMenu.name
+							);
+						} else {
+							handleUpdateMenu( navPostOrClassicMenu?.id );
```

Comment:
Can `navPostOrClassicMenu` be null? If yes, then the hasOwnProperty call above should take that into account.